mediacloud/mediawords/util/web/user_agent/test_throttled.py

```diff
@@ -0,0 +1,62 @@
+"""test ThrottledUserAgent."""
+
+import time
+
+from mediawords.test.test_database import TestDatabaseWithSchemaTestCase
+from mediawords.test.http.hash_server import HashServer
+from mediawords.util.web.user_agent.throttled import ThrottledUserAgent
+from mediawords.util.web.user_agent.throttled import McThrottledUserAgentTimeoutException
+import mediawords.util.web.user_agent.throttled
+
+
+class TestThrottledUserAgent(TestDatabaseWithSchemaTestCase):
+    """test case for ThrottledUserAgent."""
+
+    def test_request(self) -> None:
+        """Test requests with throttling."""
+        pages = {'/test': 'Hello!', }
+        port = 8888
```

Comment:
this won't work for anyone (like me!) running a Jupyter server on the default port.  You might look at https://github.com/getsentry/responses or the method used elsewhere for getting a random unused port.

Reply:
Added where? Can't find it.

Why not just use `random_unused_port()` from `mediawords.util.network`?